Competition2017/src/competition/subsystems/autonomous/DriveToHopperThenBoilerCommandGroup.java

```diff
@@ -0,0 +1,81 @@
+package competition.subsystems.autonomous;
+
+import com.google.inject.Provider;
+import competition.subsystems.drive.commands.DriveForDistanceCommand;
+import competition.subsystems.drive.commands.DriveToOrientationCommand;
+import edu.wpi.first.wpilibj.command.CommandGroup;
+import xbot.common.command.BaseCommand;
+import xbot.common.properties.BooleanProperty;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+
+public class DriveToHopperThenBoilerCommandGroup extends CommandGroup{
+    public DoubleProperty distanceFromWallToBaseline;
+    public DoubleProperty distanceFromTurningPointToHopper;
+    public DoubleProperty distanceFromTurningPointToBoiler;
+    public DoubleProperty headingToFaceHopper;
+    public DoubleProperty waitTimeForFuelCollection;
+    public DoubleProperty headingToFaceBoiler;
+    public BooleanProperty goToBoiler;
+    
+    private class TimeoutCommand extends BaseCommand {
+        @Override
+        public void initialize() {
+            this.setTimeout(waitTimeForFuelCollection.get());
+        }
+
+        @Override
+        public void execute() {}
+        
+        public boolean isFinished(){
+            return this.isTimedOut();
+        }
+    }
+    
+    public DriveToHopperThenBoilerCommandGroup(
+            XPropertyManager propManager,
+            Provider<DriveForDistanceCommand> driveForDistanceProvider, 
+            Provider<DriveToOrientationCommand> driveToOrientationProvider){
+        DriveForDistanceCommand driveForDistance = driveForDistanceProvider.get();
+        DriveToOrientationCommand driveToOrientation = driveToOrientationProvider.get();
+
+        distanceFromWallToBaseline = propManager.createPersistentProperty("distanceFromWallToBaselineInInches", 37.25);
+        headingToFaceHopper = propManager.createPersistentProperty("headingToFaceHopperInDegrees", 90);
+        distanceFromTurningPointToHopper = propManager.createPersistentProperty("distanceFromTurningPointOnBaselineToHopperInInches", 43.25);
+        waitTimeForFuelCollection = propManager.createPersistentProperty("waitTimeForFuelCollectionInSeconds", 5);
+        goToBoiler = propManager.createPersistentProperty("goToBoiler", true);
+        headingToFaceBoiler = propManager.createPersistentProperty("headingToFaceBoilerInDegees", 70.2);
+        distanceFromTurningPointToHopper = propManager.createPersistentProperty("distanceFromTurningPointToHopperInInches", 72);
+
+        // move to turning point on baseline
```

Comment:
Add a comment saying where you expect the robot to be starting, and in what orientation.